src/rgw/rgw_rest_log.cc

```diff
@@ -78,10 +78,15 @@ void RGWOp_MDLog_List::execute() {
   } 
 
   if (period.empty()) {
-    ldout(s->cct, 5) << "Missing period id" << dendl;
-    http_ret = -EINVAL;
-    return;
+    ldout(s->cct, 5) << "Missing period id trying to use current" << dendl;
+    period = store->get_current_period_id();
+    if (period.empty()) {
+      ldout(s->cct, 5) << "Missing period id" << dendl;
+      http_ret = -EINVAL;
+      return;
+    }
```

Comment:
This is a fix for the sync agent, which will always send an empty period id, right? And the sync agent is not compatible with a new multisite configuration?

If the gateway isn't running a new multisite config (i.e. realm.current_period is empty), it should just accept and use the empty period id.

But if the gateway does have a valid current_period, I think it should still reject requests that don't give a period id. The remote side needs to be explicit about which period it's syncing, so falling back to the current_period is probably not the right thing to do.

So I would suggest this change instead:
```c++
// reject empty period ids if we have a current_period
if (period.empty() && !store->get_current_period_id().empty()) {
  ldout(s->cct, 5) << "Missing period id" << dendl;
  http_ret = -EINVAL;
  return;
}
```